mleap-core/src/main/scala/ml/combust/mleap/core/feature/StringIndexerModel.scala

```diff
@@ -16,28 +16,29 @@ case class StringIndexerModel(labels: Seq[String],
                               handleInvalid: HandleInvalid = HandleInvalid.Error) extends Model {
   val stringToIndex: Map[String, Int] = labels.zipWithIndex.toMap
   private val keepInvalid = handleInvalid == HandleInvalid.Keep
-
+  private val invalidValue = labels.length
   /** Convert a string into its integer representation.
     *
     * @param value label to index
     * @return index of label
     */
   def apply(value: Any): Int = if (value == null) {
     if (keepInvalid) {
-      labels.length
+      invalidValue
```

Comment:
Neat! I didn't realize measuring a List's length was O(N) in Scala:
https://stackoverflow.com/questions/8197655/why-doesnt-the-scala-list-have-a-size-field